### ISSUE TITLE
Situation when the label is HR value instead of signal

### ISSUE THREAD

**Issue by Dylan-H-Wang** [ubicomplab/rPPG-Toolbox#200]
Hi,

Thank you for this awesome work, really amazing!

One thing I want to check is that when I tried to add new dataset, i.e., VIPL-HR-V2, and found that it seems like this toolbox only supports methods/datasets which use PPG/BVP signals as labels? Is there any suggestion if I want to use HR values directly as the ground truth? Or do you have any plan on implement this feature?

Thank you for the helps!

Kind regards,
Dylan

**Comment by yahskapar** [comment]
Hi @Dylan-H-Wang,

Can you provide a few more details on exactly what you're trying to do? Are you trying to both train and evaluate using HR labels directly? In the datasets you're considering, is there no ground truth PPG/BVP signal provided, or if it is provided is there some problem with using those labels directly instead of the HR label?

Currently the toolbox does not support using HR as a label out of the box, but in the near future we will add some support for using HR labels as a part of a loss function calculation at training time for the [PhysFormer](https://github.com/ZitongYu/PhysFormer) architecture. This will likely occur in the next two months based on my current estimate. At that point, we can also add some support for using HR labels directly at evaluation time if available. I think if you wanted to use the HR values directly, you can modify one of the trainer files (e.g., [TS-CAN trainer](https://github.com/ubicomplab/rPPG-Toolbox/blob/main/neural_methods/trainer/TscanTrainer.py) alongside making modifications to `metrics.py` and `post_process.py` as needed in the [evaluation folder](https://github.com/ubicomplab/rPPG-Toolbox/tree/main/evaluation) of the repo.

**Comment by Dylan-H-Wang** [comment]
> Hi @Dylan-H-Wang,
> 
> Can you provide a few more details on exactly what you're trying to do? Are you trying to both train and evaluate using HR labels directly? In the datasets you're considering, is there no ground truth PPG/BVP signal provided, or if it is provided is there some problem with using those labels directly instead of the HR label?
> 
> Currently the toolbox does not support using HR as a label out of the box, but in the near future we will add some support for using HR labels as a part of a loss function calculation at training time for the [PhysFormer](https://github.com/ZitongYu/PhysFormer) architecture. This will likely occur in the next two months based on my current estimate. At that point, we can also add some support for using HR labels directly at evaluation time if available. I think if you wanted to use the HR values directly, you can modify one of the trainer files (e.g., [TS-CAN trainer](https://github.com/ubicomplab/rPPG-Toolbox/blob/main/neural_methods/trainer/TscanTrainer.py) alongside making modifications to `metrics.py` and `post_process.py` as needed in the [evaluation folder](https://github.com/ubicomplab/rPPG-Toolbox/tree/main/evaluation) of the repo.
> 
> Feel free to take a look at those files and let us know if you have any questions about modifying them.

Yes, I am trying to use HR labels directly in both training and evaluation that the dataset only provides HR labels without PPG/BVP signals.

Thank you for the helps, I will try to modify your mentioned files to meet my requirement.

**Comment by yahskapar** [comment]
Going to go ahead and close this issue as it's been a while since any further discussion. Feel free to re-open or comment again with any other questions or concerns, @Dylan-H-Wang.
